src/solver/infeasible-problem-analysis/report.cpp

```diff
@@ -49,13 +49,18 @@ void InfeasibleProblemReport::buildConstraintsFromSlackVars(
 bool lessTypeName(const std::shared_ptr<WatchedConstraint> a,
                   const std::shared_ptr<WatchedConstraint> b)
 {
-    return std::type_index(typeid(*a)) < std::type_index(typeid(*b));
+    const WatchedConstraint* a_raw = a.get();
```

Comment:
Could you explain a bit more the problem we want to avoid by making this change (which seem to be equivalent) ? Thanx.

Reply:
`*a` dereferences the pointer, while `a.get()` doesn't.